var/spack/repos/builtin/packages/recorder/package.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Recorder(AutotoolsPackage):
+    """A multi-level library for understanding I/O
+    activity in HPC applications."""
+
+    homepage = "https://github.com/uiuc-hpc/Recorder"
+    url      = "https://github.com/uiuc-hpc/Recorder/archive/v2.1.4.tar.gz"
+
+    maintainers = ['wangvsa']
+
+    version('2.1.4', sha256='f66756595a7f310929c247ae03fd08a18d9843f578fffa1e3072f557bf5a158e', preferred=True)
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+    depends_on('mpi')
+    depends_on('hdf5')
```

Comment:
These can be made conditional on whether or not the variants are enabled:
```suggestion
    depends_on('mpi', when='+mpi')
    depends_on('hdf5', when='+hdf5')
```

Reply:
Recorder actually requires MPI and HDF5 even without +mpi/+hdf5.
-mpi/-hdf5 mean we do not write traces for these two levels to the file but we still intercept them.

Reply:
In that case you should probably add a comment to this effect, most packages with variants like this don't need the dependencies unless the variant is activated.

Reply:
Understood.